src/main/java/io/jenkins/plugins/ml/IPythonBuilder.java

```diff
@@ -87,7 +87,16 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launc
 
                     // Run builder on selected notebook
                     String extension = filePath.substring(filePath.lastIndexOf(".") + 1);
-                    FileExtension ext = extension.equals("ipynb") ? FileExtension.ipynb : FileExtension.json;
+                    FileExtension ext;
+                    if (extension.equals("ipynb")) {
+                        ext = FileExtension.ipynb;
+                    } else if (extension.equals("json")) {
+                        ext = FileExtension.json;
+                    } else if (extension.equals("py")) {
+                        ext = FileExtension.py;
+                    } else {
+                        ext = FileExtension.txt;
+                    }
```

Comment:
I think we can replace this `if/elseif/else` by one call to [`.valueOf`](https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#valueOf(java.lang.Class,%20java.lang.String)) @Loghijiaha (sorry, don't have the IDE to confirm or share a code snippet). If you want to have a default value, just catch the exception and set ext to text I think. Hope that makes sense.

Reply:
That will keep the code clean. :+1: